controllers/controller/workspacerouting/predictes.go

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package workspacerouting
+
+import (
+	"errors"
+
+	controllerv1alpha1 "github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+var routingPredicates = predicate.Funcs{
+	CreateFunc: func(ev event.CreateEvent) bool {
+		obj, ok := ev.Object.(*controllerv1alpha1.WorkspaceRouting)
+		if !ok {
+			return true
```

Comment:
Please add a comment that describes that it's a change related to Service/Route/Ingress related to routing. So, triggering reconcile loop which will be processed or ignored by controller...

Will it be really ignored?... I just don't understand in which place, so I need to play with it.

Reply:
I got a failed workspace once I manually created Service with manually set ownerRef, I believe it means that external routing is not going to work even with these changes.

To reproduce:
```
# do make run in another tab

# Change it when you run in another namespace
NAMESPACE=devworkspace-controller

kubectl apply -f ./samples/cloud-shell.yaml
sleep 2
DW_ID=$(kubectl get dw cloud-shell -o=yaml | yq '.status.workspaceId' -r)

# kubectl patch dw/cloud-shell --patch '{"spec":{"routingClass":"external"}}'
# ^ for some reason returns Error from server (UnsupportedMediaType): the body of the request was in an unknown format - accepted media types include: application/json-patch+json, application/merge-patch+json, application/apply-patch+yaml
# as well as 
#kubectl patch dw/cloud-shell --patch '
#spec:
#  routingClass: openshift-oauth
#'

# I'm smart!!!
kubectl get dw cloud-shell -o=yaml | sed 's/routingClass: openshift-oauth/routingClass: external/' | kubectl apply -f -
# -D

ROUTING_UID=$(kubectl get workspaceroutings.controller.devfile.io routing-${DW_ID} -o=yaml | yq '.metadata.uid' -r)


cat <<EOF | KUBECONFIG=/tmp/devworkspace-controller/kubeconfig kubectl apply -f -
apiVersion: v1
kind: Service
metadata:
  labels:
    controller.devfile.io/workspace_id: ${DW_ID}
  name: ${DW_ID}-service
  namespace: devworkspace-controller
  ownerReferences:
  - apiVersion: controller.devfile.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: WorkspaceRouting
    name: routing-${DW_ID}
    uid: ${ROUTING_UID}
spec:
  ports:
  - name: cloud-shell-proxy
    port: 4400
    protocol: TCP
    targetPort: 4400
  selector:
    controller.devfile.io/workspace_id: workspace7126ed1fe35c47f5
EOF
sleep 1
kc get dw cloud-shell -o=yaml | yq '.status'
#expected workspace is in starting phase
#actual it's failed
```

Reply:
Fixed by https://github.com/devfile/devworkspace-operator/pull/219/commits/0c70133bee407b27fa4277bec3348ea80069c086 -- turns out my reasoning was wrong in this thread: https://github.com/devfile/devworkspace-operator/pull/216#discussion_r536228115